packages/airnode-protocol/contracts/psp/BeaconServer.sol

```diff
@@ -0,0 +1,435 @@
+    {
+        require(updateRequester != address(0), "Update requester zero");
+        sponsorToUpdateRequesterToPermissionStatus[msg.sender][
+            updateRequester
+        ] = status;
+        emit SetUpdatePermissionStatus(msg.sender, updateRequester, status);
+    }
+
+    /// @notice Called to request a beacon to be updated
+    /// @dev There are two requirements for this method to be called: (1) The
+    /// sponsor must call `setSponsorshipStatus()` of AirnodeRrp to sponsor
+    /// this RrpBeaconServer contract, (2) The sponsor must call
+    /// `setUpdatePermissionStatus()` of this RrpBeaconServer contract to give
+    /// request update permission to the caller of this method.
+    /// The template used here must specify a single point of data of type
+    /// `int256` to be returned because this is what `fulfill()` expects.
+    /// This point of data must be castable to `int224`.
+    /// @param templateId Template ID of the beacon to be updated
+    /// @param sponsor Sponsor whose wallet will be used to fulfill this
+    /// request
+    /// @param sponsorWallet Sponsor wallet that will be used to fulfill this
+    /// request
+    function requestBeaconUpdate(
+        bytes32 templateId,
+        address sponsor,
+        address sponsorWallet
+    ) external override {
+        require(
+            sponsorToUpdateRequesterToPermissionStatus[sponsor][msg.sender],
+            "Caller not permitted"
+        );
+        bytes32 requestId = airnodeRrp.makeTemplateRequest(
+            templateId,
+            sponsor,
+            sponsorWallet,
+            address(this),
+            this.fulfill.selector,
+            ""
+        );
+        requestIdToTemplateId[requestId] = templateId;
+        emit RequestedBeaconUpdate(
+            templateId,
+            sponsor,
+            msg.sender,
+            requestId,
+            sponsorWallet
+        );
+    }
+
+    /// @notice Called by AirnodeRrp to fulfill the request
+    /// @dev It is assumed that the fulfillment will be made with a single
+    /// point of data of type `int256`
+    /// @param requestId ID of the request being fulfilled
+    /// @param data Fulfillment data (a single `int256` encoded as `bytes`)
+    function fulfill(bytes32 requestId, bytes calldata data)
+        external
+        override
+        onlyAirnodeRrp
+    {
+        bytes32 templateId = requestIdToTemplateId[requestId];
+        require(templateId != bytes32(0), "No such request made");
+        delete requestIdToTemplateId[requestId];
+        (int256 decodedData, uint256 timestamp) = abi.decode(
+            data,
+            (int256, uint256)
+        );
+        require(
+            decodedData >= type(int224).min && decodedData <= type(int224).max,
+            "Value typecasting error"
+        );
+        require(timestamp <= type(uint32).max, "Timestamp typecasting error");
+        require(block.timestamp - 1 hours < timestamp, "Fulfillment stale");
```

Comment:
We probably want the `1 hour` to be configurable, don't we?

Reply:
Also, do we really want to revert here or should we just emit an event and return to ignore the fulfillment tx. 

I guess the question is: does Airnode care about this require condition when fulfilling a request?

Reply:
I think 1 hour is reasonable (confirmation time under ideal conditions is <5 mins so this should be a proportionate number) and it's fine to hardcode it. Otherwise, it's not clear who should be able to configure it. We can initialize it at the constructor but that means 1 additional storage read for fulfillments = +2000 gas so I'm not sure if that's worth it.

Requiring these conditions here means `callSuccess` [here](https://github.com/api3dao/airnode/blob/master/packages/airnode-protocol/contracts/rrp/AirnodeRrp.sol#L253) will return `false` and AirnodeRrp will emit `FailedRequest` instead of `FulfilledRequest`. It feels like that's the correct thing to do.